packages/analytics-core/src/plugins/destination.ts

```diff
@@ -126,17 +124,14 @@ export class Destination implements DestinationPlugin {
       return;
     }
 
-    const list: Context[] = [];
-    const later: Context[] = [];
-    this.queue.forEach((context) => (context.timeout === 0 ? list.push(context) : later.push(context)));
-    this.queue = later;
-
+    const list = this.queue.filter((context) => context.timeout === 0);
```

Comment:
Will we fix this separately in a different PR where we'll make sure we send data in the right order?